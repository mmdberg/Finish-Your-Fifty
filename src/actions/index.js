export const captureUser = (user) => ({
  type: 'CAPTURE_USER',
  user
});

export const addRace = (race) => ({
  type: 'ADD_RACE',
  race
});

export const logOut = () => ({
  type: 'LOG_OUT'
});

export const removeRace = (race) => ({
  type: 'REMOVE_RACE',
  race
});