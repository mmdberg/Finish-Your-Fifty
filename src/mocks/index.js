export const mockRace = {
  venue: 'Atlas Park',
  city: 'San Diego',
  event: 'Spooky 5k',
  date: '04-05-2018'
};

export const mockApiResult = {
  place: {
    placeName: 'Atlas Park',
    cityName: 'San Diego'
  },
  activityStartDate: '2018-04-05T08:00:00',
  organization: {
    organizationName: 'Spooky 5k'
  }
};

export const mockUser = {
  name: 'Pizza',
  email: 'pizza@pizza',
  password: 'p'
};