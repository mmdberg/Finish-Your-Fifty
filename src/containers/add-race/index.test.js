import { shallow } from 'enzyme';
import React from 'react';
import { AddRace, mapStateToProps, mapDispatchToProps } from './index';
import { mockCompletedRace } from '../../mocks';
import * as actions from '../../actions';

describe('AddRace', () => {
  let wrapper;
  let mockAddRace = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<AddRace addRace={mockAddRace}/>);
  });

  it.skip('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should start with empty state', () => {
    const expected = {
      event: '',
      distance: '',
      time: '',
      city: '',
      state: ''
    };
    expect(wrapper.state()).toEqual(expected);
  });

  it('should update state with user input', () => {
    const mockEvent =  {target: {name: 'time', value: '45'}};
    const expected = {
      event: '',
      distance: '',
      time: '45',
      city: '',
      state: ''
    };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });

  it('should call addRace with right params on submit', () => {
    const mockEvent = { preventDefault: jest.fn()};
    wrapper.setState(mockCompletedRace);
    wrapper.instance().handleSubmit(mockEvent);
    expect(mockAddRace).toHaveBeenCalledWith(mockCompletedRace);
  });

  it('should reset state after submit', () => {
    const mockEvent = { preventDefault: jest.fn()};
    wrapper.setState(mockCompletedRace);
    const expected = {
      event: '',
      distance: '',
      time: '',
      city: '',
      state: ''
    };
    wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });

  describe('mapStateToProps', () => {
    it.skip('should correctly map races to props', () => {
      let mockState = {
        races: [mockCompletedRace]
      };
      let mapped = mapStateToProps(mockState);
      expect(mapped.races).toEqual([mockCompletedRace]);
    });
  });

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let mapped;

    beforeEach(() => {
      mockDispatch = jest.fn();
      mapped = mapDispatchToProps(mockDispatch);
    });

    it('should call dispatch with right params on addRace', () => {
      const expected = actions.addRace(mockCompletedRace);
      mapped.addRace(mockCompletedRace);
      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });
  });
});