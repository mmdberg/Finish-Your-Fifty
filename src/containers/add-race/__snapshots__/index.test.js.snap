// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRace should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddRace
    addRace={[Function]}
    user={
        Object {
            "email": "pizza@pizza",
            "id": 23,
            "password": "p",
            "userName": "Pizza",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Add a race to your log:
</h2>,
        <input
          name="raceName"
          onChange={[Function]}
          placeholder="Race Name"
          type="text"
          value=""
/>,
        <input
          name="distance"
          onChange={[Function]}
          placeholder="Distance"
          type="text"
          value=""
/>,
        <input
          name="time"
          onChange={[Function]}
          placeholder="Time"
          type="text"
          value=""
/>,
        <input
          name="city"
          onChange={[Function]}
          placeholder="City"
          type="text"
          value=""
/>,
        <input
          name="state"
          onChange={[Function]}
          placeholder="State"
          type="text"
          value=""
/>,
        <input
          checked={true}
          id="choiceTrue"
          name="completed"
          onChange={[Function]}
          type="radio"
          value="true"
/>,
        <label
          htmlFor="choiceTrue"
>
          Completed
</label>,
        <input
          checked={false}
          id="choiceFalse"
          name="completed"
          onChange={[Function]}
          type="radio"
          value="false"
/>,
        <label
          htmlFor="choiceFalse"
>
          Interested
</label>,
        <button
          type="submit"
>
          Submit
</button>,
        <p />,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a race to your log:",
        },
        "ref": null,
        "rendered": "Add a race to your log:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "raceName",
          "onChange": [Function],
          "placeholder": "Race Name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "distance",
          "onChange": [Function],
          "placeholder": "Distance",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "time",
          "onChange": [Function],
          "placeholder": "Time",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "city",
          "onChange": [Function],
          "placeholder": "City",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "state",
          "onChange": [Function],
          "placeholder": "State",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": true,
          "id": "choiceTrue",
          "name": "completed",
          "onChange": [Function],
          "type": "radio",
          "value": "true",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Completed",
          "htmlFor": "choiceTrue",
        },
        "ref": null,
        "rendered": "Completed",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "id": "choiceFalse",
          "name": "completed",
          "onChange": [Function],
          "type": "radio",
          "value": "false",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Interested",
          "htmlFor": "choiceFalse",
        },
        "ref": null,
        "rendered": "Interested",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Add a race to your log:
</h2>,
          <input
            name="raceName"
            onChange={[Function]}
            placeholder="Race Name"
            type="text"
            value=""
/>,
          <input
            name="distance"
            onChange={[Function]}
            placeholder="Distance"
            type="text"
            value=""
/>,
          <input
            name="time"
            onChange={[Function]}
            placeholder="Time"
            type="text"
            value=""
/>,
          <input
            name="city"
            onChange={[Function]}
            placeholder="City"
            type="text"
            value=""
/>,
          <input
            name="state"
            onChange={[Function]}
            placeholder="State"
            type="text"
            value=""
/>,
          <input
            checked={true}
            id="choiceTrue"
            name="completed"
            onChange={[Function]}
            type="radio"
            value="true"
/>,
          <label
            htmlFor="choiceTrue"
>
            Completed
</label>,
          <input
            checked={false}
            id="choiceFalse"
            name="completed"
            onChange={[Function]}
            type="radio"
            value="false"
/>,
          <label
            htmlFor="choiceFalse"
>
            Interested
</label>,
          <button
            type="submit"
>
            Submit
</button>,
          <p />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a race to your log:",
          },
          "ref": null,
          "rendered": "Add a race to your log:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "raceName",
            "onChange": [Function],
            "placeholder": "Race Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "distance",
            "onChange": [Function],
            "placeholder": "Distance",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "time",
            "onChange": [Function],
            "placeholder": "Time",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "city",
            "onChange": [Function],
            "placeholder": "City",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "state",
            "onChange": [Function],
            "placeholder": "State",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "id": "choiceTrue",
            "name": "completed",
            "onChange": [Function],
            "type": "radio",
            "value": "true",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Completed",
            "htmlFor": "choiceTrue",
          },
          "ref": null,
          "rendered": "Completed",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "id": "choiceFalse",
            "name": "completed",
            "onChange": [Function],
            "type": "radio",
            "value": "false",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Interested",
            "htmlFor": "choiceFalse",
          },
          "ref": null,
          "rendered": "Interested",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
